lib/rules/object-property-newline.js

```diff
@@ -56,10 +100,36 @@ module.exports = {
                 }
 
                 for (let i = 1; i < node.properties.length; i++) {
-                    const lastTokenOfPreviousProperty = sourceCode.getLastToken(node.properties[i - 1]);
-                    const firstTokenOfCurrentProperty = sourceCode.getFirstToken(node.properties[i]);
+                    const lastTokenOfPreviousProperty = sourceCode.getLastToken(node.properties[i - 1]),
+                        firstTokenOfCurrentProperty = sourceCode.getFirstToken(node.properties[i]),
+                        previousPropertyEndLine = lastTokenOfPreviousProperty.loc.end.line,
+                        currentPropertyStartLine = firstTokenOfCurrentProperty.loc.start.line,
+                        currentPropertyIsOnNewLine = previousPropertyEndLine < currentPropertyStartLine;
+                    let currentPropertyIsValid;
+
+                    if (!allowOpenBracket && !denyCommaFirst) {
+                        currentPropertyIsValid = currentPropertyIsOnNewLine;
+                    } else {
+
+                        if (allowOpenBracket && !denyCommaFirst) {
+                            currentPropertyIsValid =
+                                currentPropertyIsOnNewLine ||
+                                validPerOpenBracket(firstTokenOfCurrentProperty, sourceCode, currentPropertyStartLine);
+                        } else if (!allowOpenBracket && denyCommaFirst) {
+                            currentPropertyIsValid =
+                                currentPropertyIsOnNewLine &&
+                                validPerLeadingComma(firstTokenOfCurrentProperty, sourceCode, currentPropertyStartLine);
+                        } else if (allowOpenBracket && denyCommaFirst) {
+
+                            // A line containing only “, [” is valid here.
```

Comment:
Why should this be valid? Is that how JSCS handles this case? Would it make more sense to augment `comma-style` with an option to allow computed properties after commas in "last" mode?

Reply:
Yes, JSCS treats such a line as valid per its `requireObjectKeysOnNewLine` rule. One might presume that if JSCS has such a rule then there is some demand for it.

Given that ESLint wants to establish JSCS compatibility, it seems to me more user-friendly to tell JSCS users how to do that for a single JSCS rule with instructions for configuring a single ESLint rule, rather than making them piece together 2 ESLint rules.

Also, adding an option to `comma-style` could cover this case but not the case of a line with a lone comma on it. That would need another new option.

Also, this may be due to my inadequate knowledge, but I don’t understand how the `comma-style` rule can be simultaneously applied oppositely to 2 or more of the 10 node types that it applies to. If my impression is correct, then it seems questionable to load additional options onto a rule that already has so many. 

I have added what I think is a thorough explanation of the (in)applicability of `comma-style` to the `noCommaFirst` section of `docs/rules/object-property-newline`. I hope it at least shows why `comma-style` as it now exists can’t do the job. The other points above lead me to believe that confining these alterations to `object-property-newline` is, on balance, wiser than giving part of the job to `comma-style`, but there may be reasons for the latter that I’m not aware of.